server/pulp/server/managers/auth/cert/cert_generator.py

```diff
@@ -161,42 +162,20 @@ def decode_admin_user(self, encoded_string):
 
 class SerialNumber:
 
-    PATH = config.config.get('security', 'serial_number_path')
-    __mutex = RLock()
     __metaclass__ = Singleton
 
     def next(self):
         """
         Get the next serial#
         @return: The next serial#
-        @rtype: int
+        @rtype: str
         """
-        self.__mutex.acquire()
-        try:
-            fp = open(self.PATH, 'a+')
-            try:
-                sn = int(fp.read()) + 1
-            except:
-                sn = 1
-            fp.seek(0)
-            fp.truncate(0)
-            fp.write(str(sn))
-            fp.close()
-            return sn
-        finally:
-            self.__mutex.release()
-
-    def reset(self):
-        """
-        Reset the serial number
-        """
-        self.__mutex.acquire()
-        try:
-            fp = open(self.PATH, 'w')
-            fp.write('0')
-            fp.close()
-        finally:
-            self.__mutex.release()
+        # Generate a random 32 bit number
+        random_sn = str(getrandbits(32))
+        # Current utc time
+        curr_time = str(int(time.time()))
+
+        return curr_time + random_sn
```

Comment:
A comment explaining why it's okay for a method named "serial" to return something that is not a monotonically increasing integer would probably be helpful.